pandas/tests/io/test_excel.py

```diff
@@ -2490,3 +2499,71 @@ def custom_converter(css):
             n_cells += 1
 
     assert n_cells == (10 + 1) * (3 + 1)
+
```

Comment:
Not sure I'm getting you entirely, as I'm trying to work out how this relates to the boundaries between io/excel and io/formats/excel. Do you want to see this all merged?

IMO, a logical structure for the code might be:

* `pandas/io/excel/read.py` - `*Reader`
* `pandas/io/excel/write.py` - `ExcelFormatter`, `*Writer`
* `pandas/io/excel/style.py` - `CSSToExcelConverter`

with test files to correspond.

Or perhaps `pandas/io/excel/format.py` should include `CSSToExcelConverter` and `ExcelFormatter` as `pandas/io/formats/excel.py` does currently.

However, the tests you're concerned to move here are end-to-end tests covering the `Styler.to_excel` interface and writers.

Please clarify your vision if you can.